internal/state/config/config.go

```diff
@@ -0,0 +1,177 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	toml "github.com/pelletier/go-toml/v2"
+)
+
+type Config interface {
+	Write() error
+	Marshal() ([]byte, error)
```

Comment:
This is only called from `Config.Write()` and can be removed from the interface.

Reply:
What do you think about removing `ContextNames`, `ContextByName` and `RemoveContext` too? Should they stay mockable or can we just make a utility function that takes a `config.Config` and implements the functionality using `Contexts` and `SetContexts`?

Reply:
Fine by me, if it makes the code cleaner

Reply:
Done in 6292032 and 68a7918